index.js

```diff
@@ -84,6 +84,13 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
+  if (typeof name !== 'string') {
+    throw new TypeError('argument name must be a string');
+  }
+  if (typeof val !== 'string') {
```

Comment:
I have to think on this new restriction a little bit, because it's a breaking change since it currently works just fine with non-strings (especially numbers), which people are likely to be providing currently.